src/components/component.ts

```diff
@@ -356,15 +360,17 @@ export class Component {
 
   private _generateClipPath() {
     // The clip path will prevent content from overflowing its Component space.
-    // HACKHACK: IE <=9 does not respect the HTML base element in SVG.
-    // They don't need the current URL in the clip path reference.
-    let prefix = /MSIE [5-9]/.test(navigator.userAgent) ? "" : document.location.href;
-    prefix = prefix.split("#")[0]; // To fix cases where an anchor tag was used
     let clipPathId = Utils.DOM.generateUniqueClipPathId();
-    this._element.attr("clip-path", "url(\"" + prefix + "#" + clipPathId + "\")");
-    let clipPathParent = this._boxContainer.append("clipPath")
-                                           .attr("id", clipPathId);
+    let clipPathParent = this._boxContainer.append("clipPath").attr("id", clipPathId);
     this._addBox("clip-rect", clipPathParent);
+    this._updateClipPath = () => {
+      // HACKHACK: IE <= 9 does not respect the HTML base element in SVG.
+      // They don't need the current URL in the clip path reference.
+      let prefix = /MSIE [5-9]/.test(navigator.userAgent) ? "" : document.location.href;
+      prefix = prefix.split("#")[0]; // To fix cases where an anchor tag was used
+      this._element.attr("clip-path", "url(\"" + prefix + "#" + clipPathId + "\")");
+    };
```

Comment:
I guess my main concern is that, since `_updateClipPath` is defined at runtime, developers will need to consider the case if that function is actually defined by the time it runs into their code path, meaning the need to have existence checks for this function.

That said, it seems that the clipPath code is so one-off that the scope of defining and checking for this functions is pretty narrow.  We can run with this for now and hopefully it doesn't bite us.